File: problems/problems_238/solution.go
package problem238

import (
	"encoding/json"
	"log"
	"strings"
)

func productExceptSelf(nums []int) []int {
	n := len(nums)
	res := make([]int, n)
	res[0] = 1
	for i := 1; i < n; i++ {
		res[i] = res[i-1] * nums[i-1]
	}
	right := 1
	for i := n - 1; i >= 0; i-- {
		res[i] *= right
		right *= nums[i]
	}
	return res
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return productExceptSelf(nums)
}

File: problems/problems_238/solution.ts
function productExceptSelf(nums: number[]): number[] {
    const n: number = nums.length;
	const ans: number[] = new Array(n).fill(1);
	for (let i: number = 1; i < n; i++) {
		ans[i] = ans[i - 1] * nums[i - 1];
	}
	let right: number = 1;
	for (let i: number = n - 1; i >= 0; i--) {
		ans[i] *= right;
		right *= nums[i];
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return productExceptSelf(nums);
}
